### ISSUE TITLE
Calendar Catchup

### ISSUE THREAD

**Issue by alexec** [argoproj/argo-events#793]
**Is your feature request related to a problem? Please describe.**

Users want to be able to pretend that a calendar was running in the past and produces the events it would have. E

**Describe the solution you'd like**

Some way to configure the calendar to include the backfill information

```yaml
backfill:
  from: date
  to: date
```

And some way to determine if we have already run the backfill:

```yaml
status:
  backfill:
    lastDate: date
```

Note -  we cannot store this status in the event source resource as it must be in user namespace.

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: same-name-as-resource  
data:
  calender.backfill.lastDate: date
```

**Describe alternatives you've considered**

This could be a feature in Argo Cron workflows.

**Additional context**

We need to determine how to store executions.

We can generate them as follows:

https://github.com/robfig/cron/issues/333
---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by alexec** [comment]
After a bit of thought, I think we should only have one date `last` and store that in a config map in the user namespace. This indicates the last time a calendar emitted an event. After any crash we can start from that point again.  It can also be set far back in time to do backfill.

**Comment by alexec** [comment]
https://medium.com/@ankur.choraywal/airflow-backfill-plugin-1c8cc916e8bc&sa=D&ust=1600142635037000&usg=AFQjCNH3sasrfXo5Q1kH608jksudJSV36Q

**Comment by alexec** [comment]
@whynowy we have enough information to start on this.

As part of getting the rest of the team more up to speed, would it be good for @sarabala1979 or @simster7 to take this on?

**Comment by alexec** [comment]
@sarabala1979 done?

**Comment by sarabala1979** [comment]
@alexec This is actually a Catchup feature and it is done and released in v1.1.0
#897 dup
